src/utils.ts

```diff
@@ -0,0 +1,107 @@
+import { watch, ref } from 'vue';
+import type { Ref } from 'vue';
+import { useApi, useStores } from '@directus/extensions-sdk';
+
+export function checkFieldInTemplate(template: string, field: string) {
+	const matches = template.match(/{{.*?}}/g);
+	return (matches || []).some((m) => m.includes(field));
+}
+
+const checkFieldNeedsUpdate = (field: string, newVal: Record<string, any>, oldVal: Record<string, any>) => {
+	if (oldVal[field] === newVal[field]) {
+		return false;
+	}
+	if (JSON.stringify(oldVal[field]) === JSON.stringify(newVal[field])) {
+		return false;
+	}
+	return true;
+};
+
+export const useCollectionRelations = (collection: string): Ref<any[]> => {
+	const { useRelationsStore } = useStores();
+	const { getRelationsForCollection } = useRelationsStore();
+	return ref(getRelationsForCollection(collection));
+};
+
+interface IRelationUpdate<T = unknown> {
+	create: T[];
+	update: { owner: string; id: number | string }[];
+	delete: (number | string)[];
+}
+
+export const useDeepValues = (
+	values: Ref<Record<string, any>>,
+	relations: Ref<any[]>,
+	collection: string,
+	pk: string,
+	template: string
+) => {
+	const api = useApi();
+	const finalValues = ref<Record<string, any>>(values.value);
+	watch(values, async (val, oldVal) => {
+		const relationalData: Record<string, any> = {};
+
+		for (const key of Object.keys(values.value)) {
+			const relation = relations.value.find((rel) => rel.meta?.one_field === key);
+
+			if (!relation || !checkFieldInTemplate(template, key) || !checkFieldNeedsUpdate(key, val, oldVal)) {
+				continue;
+			}
+
+			const fieldName = relation.meta.one_field;
+			const fieldChanges = values.value[fieldName] as IRelationUpdate;
+
+			if (!fieldChanges) {
+				continue;
+			}
+
+			let arrayOfIds: (string | number)[] = [];
+			let arrayOfData: unknown[] = [];
+			if (pk !== '+') {
+				const {
+					data: { data },
+				} = await api.get(`items/${collection}/${pk}`, {
+					params: {
+						fields: [fieldName],
+					},
+				});
+				arrayOfIds = arrayOfIds.concat(data[key]);
+			}
+
+			if (fieldChanges.update) {
+				arrayOfIds = arrayOfIds.concat(fieldChanges.update.map(({ id }) => id));
+			}
+
+			if (fieldChanges.delete) {
+				arrayOfIds = arrayOfIds.filter((id) => !fieldChanges.delete.includes(id));
+			}
+
+			if (arrayOfIds.length) {
+				const {
+					data: { data },
+				} = await api.get(`items/${relation.collection}`, {
+					params: { filter: { id: { _in: arrayOfIds } } },
+				});
+
+				// merging item updates
+				arrayOfData = data.map((item: any) => ({
+					...item,
+					...fieldChanges.update?.find(({ id }) => item.id === id),
+				}));
+			}
+
+			// must concat after request, created items doenst have ids
+			if (fieldChanges.create) {
+				arrayOfData = arrayOfData.concat(fieldChanges.create);
+			}
+
+			relationalData[key] = arrayOfData;
+		}
+
+		if (Object.keys(relationalData).length) {
+			finalValues.value = { ...values.value, ...relationalData };
```

Comment:
If `relationalData` is empty, we should still assign `finalValues` to the new `values.value`. Otherwise, any non-relational field used in the template won't trigger an update.

Reply:
The problem here is: when our computed field computes and update your value, this `watch` effect will be triggered again, on that time, our relational field on template will not recalculate, because our validation on `checkFieldNeedsUpdate()`. So, if we assign the values again, the relational final value will be something like: `{ create: [], update: [], delete: []}` again, then our `ASUM` operation will complains that `a.reduce` is not a function. 
I'm trying to find a way to avoid this, maybe preventing the watch effect if our computed field changes. 

Will commit when I find a solution. Suggestions are welcome xD

Reply:
How about placing the `shouldUpdate` function in the `watch` inside `useDeepValues` and removing one from the `watch` in interface.vue?

It should solve both cases. Even though there will be redundant API calls if the non-relational field triggers the watch, I think it can be accepted. So just leave them be, for now.

Reply:
My last commit solved. 
However, we are making all the requests logic two times per relational update, can't fix it.
 > So just leave them be, for now.
 
 Agreed!